metals/src/main/scala/scala/meta/internal/parsing/TokenEditDistance.scala

```diff
@@ -96,6 +99,73 @@ sealed trait TokenEditDistance {
     }
   }
 
+  final def toRevised(
+      textEdit: l.TextEdit,
+      adjustWithinToken: Boolean,
+      fallbackToNearest: Boolean,
+  ): Option[l.TextEdit] = {
+    val revisedRange = toRevised(
+      range = textEdit.getRange(),
+      adjustWithinToken = adjustWithinToken,
+      fallbackToNearest = fallbackToNearest,
+    )
+    revisedRange.map(r => new l.TextEdit(r, textEdit.getNewText()))
+  }
+
+  final def toRevised(
+      scalaAction: b.ScalaAction,
+      adjustWithinToken: Boolean,
+      fallbackToNearest: Boolean,
+  ): Option[b.ScalaAction] = {
+    val existingEdits = scalaAction.asLspTextEdits
+    existingEdits
+      .map { textEdit =>
+        toRevised(
+          textEdit = textEdit,
+          adjustWithinToken = adjustWithinToken,
+          fallbackToNearest = fallbackToNearest,
+        )
+      }
+      .foldLeft[Option[Seq[l.TextEdit]]](Some(Seq.empty)) {
```

Comment:
This is `traverse`, though I didn't see it defined anywhere or pulled in from a 3rd party lib :sweat_smile: